components/LocationList.js

```diff
@@ -1,22 +1,71 @@
 import Card from "./Card";
-import styled from "styled-components";
-import Link from "next/link";
-import { StyledSection } from "./StyledSection";
+import { StyledSection, StyledRowSection } from "./StyledSections";
+import { StyledEditButton } from "./StyledButtons";
+import useLocalStorageState from "use-local-storage-state";
+import { FormContainer, Label, Input } from "./StyledFormElements";
+import { StyledEditModal } from "./StyledEditModal";
+import { StyledLink } from "./StyledLinks";
 
-export const StyledLink = styled(Link)`
-  text-decoration: none;
-`;
+export default function LocationList({ locations, setLocations }) {
```

Comment:
you normally want to write handler functions that update a state in the same file where the state is initialized and pass those handler down as props to update the state.

so you would not pass down setLocations here, but a handler function that does what setLocation currently does in this file instead.

Reply:
Hi Niklas, I really tried to implement your suggestion, but I do not know how. As I initialize the locations/colonies state in the _app.js, would I need to write the handler function there? How would it have to look like? Would it have to contain what you suggested in one of the prior comments (setLocations(locations.map((location) => {... }) or something else?